File: edatamate/backend/common/src/main/java/com/edatamate/common/dataset/Dataset.java
package com.edatamate.common.dataset;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * 数据集实体类
 *
 * @author: dallas
 * @since: 2025-07-14
 */
@Setter
@Getter
@TableName("t_dataset")
public class Dataset {
    /**
     * 数据集ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 数据集名称
     */
    private String name;

    /**
     * 数据集描述
     */
    private String description;

    /**
     * 数据集类型
     */
    private DatasetType type;

    /**
     * 数据集状态
     */
    private DatasetStatus status;

    /**
     * 数据集所属父级ID
     */
    private Long parentId;

    /**
     * 数据集创建时间
     */
    private LocalDateTime createdTime;

    /**
     * 数据集创建人
     */
    private String createdBy;

    /**
     * 数据集更新时间
     */
    private LocalDateTime updatedTime;

    /**
     * 数据集更新人
     */
    private String updatedBy;
}

File: frontend/src/types/dataset.ts
 export enum DatasetType {
    PRETRAIN = 'PRETRAIN',
    FINE_TUNE = 'FINE_TUNE',
    EVAL = 'EVAL',

    PRETRAIN_TEXT = "PRETRAIN_TEXT",
    PRETRAIN_IMAGE = "PRETRAIN_IMAGE",
    PRETRAIN_AUDIO = "PRETRAIN_AUDIO",
    PRETRAIN_VIDEO = "PRETRAIN_VIDEO",

    FINE_TUNE_ALPACA = "FINE_TUNE_ALPACA",
    FINE_TUNE_CHATGLM = "FINE_TUNE_CHATGLM",
    FINE_TUNE_BLOOMZ = "FINE_TUNE_BLOOMZ",
    FINE_TUNE_LLAMA = "FINE_TUNE_LLAMA",

    EVAL_GSM8K = "EVAL_GSM8K",
    EVAL_SQUAD = "EVAL_SQUAD",
    EVAL_MNLI = "EVAL_MNLI",
    EVAL_IMDB = "EVAL_IMDB",
    EVAL_SINGLE_CHOICE_QA = "EVAL_SINGLE_CHOICE_QA",
}

export enum DatasetStatus {
    DRAFT = "DRAFT",
    PROCESSING = "PROCESSING",
    ARCHIVED = "ARCHIVED",
    PUBLISHED = "PUBLISHED",
}

export interface DatasetFile {
    id: number
    name: string
    size: string
    uploadDate: string
    path: string
}

export interface Dataset {
    id: number
    name: string
    description: string
    parentId?: number
    type: DatasetType
    status: DatasetStatus
    size?: string
    itemCount?: number
    createdBy: string
    createdTime: string
    updatedBy: string
    updatedTime: string
    lastModified: string
    tags: string[]
    quality: number
    isFavorited?: boolean
    files?: DatasetFile[]
    annotations?: {
        total: number
        completed: number
        accuracy: number
    }
    lineage?: {
        source: string
        processing: string[]
        training?: {
            model: string
            accuracy: number
            f1Score: number
        }
    }
}

export interface ScheduleConfig {
    type: "immediate" | "scheduled"
    scheduleType?: "daily" | "weekly" | "monthly" | "custom"
    time?: string
    dayOfWeek?: string
    dayOfMonth?: string
    cronExpression?: string
    maxExecutions?: number
    executionCount?: number
}

export interface DatasetTask {
    id: number
    name: string
    description: string
    type: string
    status: "importing" | "waiting" | "completed" | "failed"
    progress: number
    createdAt: string
    importConfig: any
    scheduleConfig: ScheduleConfig
    nextExecution?: string
    lastExecution?: string
    executionHistory?: { time: string; status: string }[]
}
